boards.txt

```diff
@@ -6904,6 +6904,266 @@ sparkfun_esp32s2_thing_plus.menu.EraseFlash.none.upload.erase_cmd=
+sparkfun_esp32s3_thing_plus.menu.JTAGAdapter.external.build.copy_jtag_files=1
+sparkfun_esp32s3_thing_plus.menu.JTAGAdapter.bridge=ESP USB Bridge
+sparkfun_esp32s3_thing_plus.menu.JTAGAdapter.bridge.build.openocdscript=esp32s3-bridge.cfg
+sparkfun_esp32s3_thing_plus.menu.JTAGAdapter.bridge.build.copy_jtag_files=1
+
+sparkfun_esp32s3_thing_plus.menu.PSRAM.enabled=QSPI PSRAM
+sparkfun_esp32s3_thing_plus.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM
+sparkfun_esp32s3_thing_plus.menu.PSRAM.enabled.build.psram_type=qspi
+
+sparkfun_esp32s3_thing_plus.menu.PSRAM.disabled=Disabled
+sparkfun_esp32s3_thing_plus.menu.PSRAM.disabled.build.defines=
+sparkfun_esp32s3_thing_plus.menu.PSRAM.disabled.build.psram_type=qspi
+sparkfun_esp32s3_thing_plus.menu.PSRAM.opi=OPI PSRAM
+sparkfun_esp32s3_thing_plus.menu.PSRAM.opi.build.defines=-DBOARD_HAS_PSRAM
+sparkfun_esp32s3_thing_plus.menu.PSRAM.opi.build.psram_type=opi
+
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio=QIO 80MHz
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio.build.flash_mode=dio
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio.build.boot=qio
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio.build.boot_freq=80m
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio.build.flash_freq=80m
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio120=QIO 120MHz
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio120.build.flash_mode=dio
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio120.build.boot=qio
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio120.build.boot_freq=120m
+sparkfun_esp32s3_thing_plus.menu.FlashMode.qio120.build.flash_freq=80m
+sparkfun_esp32s3_thing_plus.menu.FlashMode.dio=DIO 80MHz
+sparkfun_esp32s3_thing_plus.menu.FlashMode.dio.build.flash_mode=dio
+sparkfun_esp32s3_thing_plus.menu.FlashMode.dio.build.boot=dio
+sparkfun_esp32s3_thing_plus.menu.FlashMode.dio.build.boot_freq=80m
+sparkfun_esp32s3_thing_plus.menu.FlashMode.dio.build.flash_freq=80m
+sparkfun_esp32s3_thing_plus.menu.FlashMode.opi=OPI 80MHz
+sparkfun_esp32s3_thing_plus.menu.FlashMode.opi.build.flash_mode=dout
+sparkfun_esp32s3_thing_plus.menu.FlashMode.opi.build.boot=opi
+sparkfun_esp32s3_thing_plus.menu.FlashMode.opi.build.boot_freq=80m
+sparkfun_esp32s3_thing_plus.menu.FlashMode.opi.build.flash_freq=80m
+
+sparkfun_esp32s3_thing_plus.menu.FlashSize.4M=4MB (32Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.4M.build.flash_size=4MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.8M=8MB (64Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.8M.build.flash_size=8MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.8M.build.partitions=default_8MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.16M=16MB (128Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.16M.build.flash_size=16MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.32M=32MB (256Mb)
+sparkfun_esp32s3_thing_plus.menu.FlashSize.32M.build.flash_size=32MB
+sparkfun_esp32s3_thing_plus.menu.FlashSize.32M.build.partitions=app5M_fat24M_32MB
```

Comment:
> This version of the ESP32-S3 has 4MB Flash memory and 2MB of dynamic/PSRAM

As the board uses the ESP32-S3-MINI-1-N4R2 with 4MB flash (QUAD) and 2MB PSRAM (QUAD), those options can be removed so users are not able to select wrong configuration.

```suggestion
```

Reply:
done